backbone/src/main/java/org/researchstack/backbone/step/active/AccelerometerRecorder.java

```diff
@@ -0,0 +1,75 @@
+package org.researchstack.backbone.step.active;
+
+import android.content.Context;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+
+import com.google.gson.JsonObject;
+
+import org.researchstack.backbone.step.Step;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by TheMDP on 2/5/17.
+ *
+ * This class uses the JsonArrayDataRecorder class to save the Accelerometer sensor's data as
+ * an array of accelerometer json objects with timestamp, ax, ay, and az
+ */
+
+public class AccelerometerRecorder extends SensorRecorder {
+
+    public static final String TIMESTAMP_KEY        = "timestamp";
+    public static final String ACCELERATION_X_KEY   = "x";
+    public static final String ACCELERATION_Y_KEY   = "y";
+    public static final String ACCELERATION_Z_KEY   = "z";
+
+    private JsonObject jsonObject;
+
+    /** Default constructor for serialization/deserialization */
+    AccelerometerRecorder() {
+        super();
+    }
+
+    AccelerometerRecorder(double frequency, String identifier, Step step, File outputDirectory) {
+        super(frequency, identifier, step, outputDirectory);
+    }
+
+    @Override
+    protected List<Integer> getSensorTypeList() {
+        return Collections.singletonList(Sensor.TYPE_ACCELEROMETER);
+    }
+
+    @Override
+    public void start(Context context) {
+        super.start(context);
+        if (isRecording()) {
+            jsonObject = new JsonObject();
+        }
+    }
+
+    /**
+     * Called by the base class at the Recorder's frequency
+     */
+    @Override
+    protected void writeJsonData() {
+        jsonObject.addProperty(TIMESTAMP_KEY, System.currentTimeMillis());
```

Comment:
The "addProperty" method adds or overwrites the the current property. So this just sets the value.  Then, we write the whole json object to file in the overridden "writeJsonData()" method.